peer/peer.go

```diff
@@ -39,6 +40,26 @@ type Peer struct {
 	AuthInfo credentials.AuthInfo
 }
 
+// String ensures the Peer types implements the Stringer interface in order to
+// allow to print a context with a peerKey value effectively.
+func (p *Peer) String() string {
+	if p == nil {
+		return "Peer<nil>"
+	}
+	ret := "Peer{"
+	if p.Addr != nil {
+		ret += fmt.Sprintf("Addr: '%s'", p.Addr.String())
+	}
+	if p.AuthInfo != nil {
+		if len(ret) > 5 {
+			ret += ", "
+		}
+		ret += fmt.Sprintf("AuthInfo: '%s'", p.AuthInfo.AuthType())
```

Comment:
The `nil` checks are there to avoid risks of `nil` deref when printing/logging, which aren't avoided with the proposed alternative.

A more readable alternative would be to print all fields but put `<nil>` when these are `nil`, what do you prefer?